targets/TARGET_Samsung/TARGET_SIDK_S5JS100/sleep.c

```diff
@@ -0,0 +1,138 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <stdio.h>
+#include "sleep_api.h"
+#include "cmsis.h"
+#include "gpio_api.h"
+#include "mbed_power_mgmt.h"
+#include "mbed_critical.h"
+#include "rtos_idle.h"
+
+int nbsleep_req = 0;
+int external_pin = 0;
+static int initialize_policy = 0;
+static int disable_poweroffbycp = 1;
+
+static inline void hw_delay_us(unsigned int value)
+{
+	volatile unsigned i, j;
+	for (i = 0; i < (value * 2); i++) {
+		for (j = 0; j < 100; j++);
+	}
+}
+
+int s5js100_idle_sicd(void)
+{
+	if(getreg32(0x85023090) & (0x1 << 1)) {
+		/* send CP to ready status */
+		modifyreg32(0x85023090, (0x1 << 2) , (0x1 << 2));
+		modifyreg32(0x85026100, 0x2, 0x0);
+		pmu_short_sleep(); //go to sleep
+		modifyreg32(0x85026100, 0x2, 0x2);
+
+		/* wait for mbox */
+		while((getreg32(0x85023090) & (0x1 << 1))) {
+			hw_delay_us(10);
+			putreg32(0x2,0x85023020);
+			putreg32(0x2,0x8502301C);
+		};
+
+		/* release short sleep status */
+		modifyreg32(0x85023090, (0x1 << 2) , (0x0 << 2));
+		return 1;
+	}
+	return 0;
+}
+
+static void s5js100_idle_hook(void)
```

Comment:
Why are you redefining the idle hook? What's wrong with the default Mbed one? It's not meant for platform code to change, but for apps developers.